src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -236,6 +236,29 @@ private MessagePayloadDataConverter CreateErrorDataConverter(IErrorSerializerSet
             return new MessagePayloadDataConverter(errorSerializerSettingsFactory.CreateJsonSerializerSettings(), isDefault);
         }
 
+        private IDurabilityProviderFactory GetDurabilityProviderFactory(ILogger logger, IEnumerable<IDurabilityProviderFactory> orchestrationServiceFactories)
+        {
+            bool storageTypeIsConfigured = this.Options.StorageProvider.TryGetValue("type", out object storageType);
+            if (!storageTypeIsConfigured)
+            {
+                logger.LogInformation("Using the default storage provider: Azure Storage.");
+                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals("Azure Storage"));
+            }
+
+            if (orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals(storageType)) == null)
+            {
+                IList<string> factoryNames = orchestrationServiceFactories.Select(f => f.Name).ToList();
+                logger.LogWarning($"Storage provider type ({storageType}) was not found. " +

```

Comment:
Do we want to select the AzureStorage provider when the storageType provided doesn't actually exist, or should we throw an exception? I'm just afraid of cases where a customer accidentally typos their provider name but the program relatively silently succeeds, just with a different backend than intended.

Reply:
+1 for throwing an exception if the configuration contains an error.

Reply:
+1 for throwing an exception if the configuration contains an error.